doc/tutorial/introduction/index.md

```diff
@@ -0,0 +1,120 @@
+
+SeqAn3 provides fast and easy access to biological file formats.
+The following code example demonstrates the interface of seqan3::sequence_file_input.
+
+\snippet introduction_align.cpp sequence_input_include
+\snippet introduction_align.cpp sequence_input
+
+Can you imagine anything easier? After you have initialised the instance with a filename,
+you can simply query the data of interest using the get function. The available fields are
+`SEQ`, `ID` and `QUAL` to retrieve sequences, ids and qualities, respectively. The latter is empty
+unless you read FastQ files. The appropriate file format is detected by SeqAn from your file name suffix.
+
+Here is the content of `seq.fasta`, so you can try it out!
+
+~~~
+>seq1
+ACGTGATG
+>seq2
+AGTGATACT
+~~~
+
+<br>
+
+---
+
+# Align two sequences
+
+We have two sequences from the file above now – so let us align them.
+The pairwise sequence alignment is one of the core algorithms in SeqAn and used by several library components
+and apps. It is strongly optimised for speed and parallel execution while providing exact results and a
+generic interface.
+
+\snippet introduction_align.cpp alignment_include
+\snippet introduction_align.cpp alignment
+
+The algorithm returns a range of result objects – which is the reason for the loop here (in this case the range
+has length 1). Instead of passing a single pair of sequences, we could give a vector of sequence pairs to the
+algorithm which then executes all alignments in parallel and stores the results in various seqan3::align_result
+objects. The second argument to seqan3::align_pairwise is the *configuration* which allows you to specify
+a lot of parameters for the alignment computation, for instance score functions, banded alignment and whether
+you wish to compute a traceback or not. The configurations have their own namespace seqan3::align_cfg and can
+be piped for building combinations. Check the alignment tutorial if you want to learn more.
+
+\note
+Every directory in the SeqAn sources contains an `all.hpp` file which includes all the functionality
+of the respective (sub-) module. Thus, if you are working with several alignment methods, you can simply
+include `seqan3/alignment/all.hpp` instead of naming each file individually.
+
+<br>
+
+---
+
+# Indexed searching
+At the end of this introduction tutorial, we want to present you another efficient algorithm in SeqAn:
+The following code example shows you the steps for performing rapid searches in an indexed genome sequence.
+As index we use the FM index here. After creation we want to query the positions of the motif *TAG* inside
+a given sequence.
+
+\snippet introduction_align.cpp index_search_include
+\snippet introduction_align.cpp index_search
+
+For more information on indexing and searching we refer you to the *Index Tutorial*. We hope you got a rough
+overview on how the SeqAn3 interface has developed since version 2 and you are now free to continue with the
+tutorials of your interest.
```

Comment:
I think this Tutorial is written well, but I would have preferred something more "aus-einem-Guss", e.g. it is not clear why we do indexing after alignment. Usually it is the other way around. Also the arg-parser is not used... One program as a whole, with annotated steps would be easier, or not?

Also some crucial "meta" things are missing. I think the "first steps" should explain a little bit more about SeqAn, e.g.
  - that SeqAn3 is comprised of Modules, what these modules are and how they relate to includes. The current text is very short....
  - it should teach a little about doxygen, e.g. "If you remember the name of a function or class, but don't know which (sub-)module it belongs to, you can enter it in the search bar (top-right)."...
  - it should teach about the SeqAn style, e.g.
    - "SeqAn3 uses `snake_case` for almost everything, also classes. Only C++ concepts are named using `CamelCase`."
    - "We use a lot of Modern C++ in SeqAn3 so some things might look alien at first, e.g. type templates are used like ordinary types in many situations (no `<>`). We also always use `{}` to initialise objects and not `()` which is only used for function calls. In general the style should be much easier for newcomers."